Tests/TracingTests/LegacyBaggageTests.swift

```diff
@@ -0,0 +1,32 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Distributed Tracing open source project
+//
+// Copyright (c) 2020-2023 Apple Inc. and the Swift Distributed Tracing project
+// authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@testable import Instrumentation
+import ServiceContextModule
+import InstrumentationBaggage // legacy module, kept for easier migrations
```

Comment:
I thought we are going to pull out the legacy baggage module? Should we remove this test?

Reply:
So this module exists in the context repo and it is optional and immediately deprecated in there. It is a module with a typealias to the new context style, so if someone uses it explicitly they’ll get a migration warning. Though we don’t reexport this module anymore in tracing… 

I did not make any APIs in tracing use baggage though.

it’s a bit tricky to judge how worth it is

Reply:
IMO if we are going to drop baggage here we should also do it over in the service-context package. This way we don't release software with immediate deprecations

Reply:
Yeah I can be convinced to do this I guess hmmm... we had this idea to be less breaking to folks with @tomerd bur maybe it doesn't matter? It would only work for those that explicitly used the baggage MODULE which should be very few... like just @weissi probably hm

Reply:
Heh actually, we can't have this test since we're testing here with warnings-as-errors here so the deprecation becomes an error and fails the build 😉 

> error: 'Baggage' is deprecated: Use 'ServiceContext' from 'ServiceContextModule' instead.